Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -0,0 +1,450 @@
+        second_trans_workspace = self._sumWorkspaces(secondTransRuns, secondTransWorkspaces, True)
+        # Slice the input workspace, if required
+        input_workspace = self._sliceWorkspace(input_workspace)
+        # Perform the reduction
+        alg = self._reduce(input_workspace, first_trans_workspace, second_trans_workspace)
+        self._finalize(alg)
+
+    def validateInputs(self):
+        """Return a dictionary containing issues found in properties."""
+        issues = dict()
+        if len(self.getProperty(Prop.RUNS).value) > 1 and self.getProperty(Prop.SLICE).value:
+            issues[Prop.SLICE] = "Cannot perform slicing when summing multiple input runs"
+        return issues
+
+    def _declareSliceAlgorithmProperties(self):
+        """Copy properties from the child slicing algorithm and add our own custom ones"""
+        self.declareProperty(Prop.SLICE, False, doc = 'If true, slice the input workspace')
+        whenSliceEnabled = EnabledWhenProperty(Prop.SLICE, PropertyCriterion.IsEqualTo, "1")
+
+        self._slice_properties = ['TimeInterval', 'LogName', 'LogValueInterval']
+        self.copyProperties('ReflectometrySliceEventWorkspace', self._slice_properties)
+        for property in self._slice_properties:
+            self.setPropertySettings(property, whenSliceEnabled)
+
+        self.declareProperty(name=Prop.NUMBER_OF_SLICES,
+                             defaultValue=Property.EMPTY_INT,
+                             validator=IntBoundedValidator(lower=1),
+                             direction=Direction.Input,
+                             doc='The number of uniform-length slices to slice the input workspace into')
+        self.setPropertySettings(Prop.NUMBER_OF_SLICES, whenSliceEnabled)
+
+    def _declareReductionAlgorithmProperties(self):
+        """Copy properties from the child reduction algorithm"""
+        self._reduction_properties = [
+            'SummationType', 'ReductionType', 'IncludePartialBins',
+            'AnalysisMode', 'ProcessingInstructions', 'ThetaIn', 'ThetaLogName', 'CorrectDetectors',
+            'DetectorCorrectionType', 'WavelengthMin', 'WavelengthMax', 'I0MonitorIndex',
+            'MonitorBackgroundWavelengthMin', 'MonitorBackgroundWavelengthMax',
+            'MonitorIntegrationWavelengthMin', 'MonitorIntegrationWavelengthMax',
+            'NormalizeByIntegratedMonitors', 'Params', 'StartOverlap', 'EndOverlap',
+            'CorrectionAlgorithm', 'Polynomial', 'C0', 'C1',
+            'MomentumTransferMin', 'MomentumTransferStep', 'MomentumTransferMax',
+            'PolarizationAnalysis', 'CPp', 'CAp', 'CRho', 'CAlpha', 'FloodCorrection',
+            'FloodWorkspace', 'Debug']
+        self.copyProperties('ReflectometryReductionOneAuto', self._reduction_properties)
+
+    def _getInputWorkspaces(self, runs, isTrans):
+        """Convert the given run numbers into real workspace names. Uses workspaces from
+        the ADS if they exist, or loads them otherwise."""
```

Comment:
Maybe explore `LoadAndMerge`. It uses `MergeRuns`.

It could be used instead of `_sumWorkspaces` and `_loadRun`

Here is how we use it
```
            loadOption = {
                'XUnit': 'TimeOfFlight',
                'BeamCentre': 127.5,
                'BraggAngle': 0.0
            }
            # MergeRunsOptions are defined by the parameter files and will not be modified here!
            ws = LoadAndMerge(
                Filename=inputFiles,
                LoaderName='LoadILLReflectometry',
                LoaderVersion=-1,
                LoaderOptions=loadOption,
                OutputWorkspace=mergedWSName,
                EnableLogging=self._subalgLogging

```

Reply:
`MergeRuns` looks better than `Plus` :+1:

Again, I don't think I can use `LoadAndMerge` because the input might be a list of workspace names.